BOOST_REGEX_UNICODE_RUNS check fails with changes introduced in a87acf2c3be27cb3fdc9484f38a8f4ef46e08f29

@jwiegley it seems the changes introduced in https://github.com/ledger/ledger/commit/a87acf2c3be27cb3fdc9484f38a8f4ef46e08f29 break the `BOOST_REGEX_UNICODE_RUNS` check in CMakeLists.txt.

I'm uncertain what the best way forward is, from a conversation with ChatGPT a _"UTF-8 string literal [cannot be] directly [assigned] to a `std::basic_string<boost::uint32_t>`"_. Which suggests that ledger's code in this regard would benefit from a modernisation. Unfortunately I understand too little about the issue.

MWE:

```
% cat boost_regex_unicode_runs.cpp
#include <boost/regex/icu.hpp>

using namespace boost;

int main() {
  std::string text = "Активы";
  std::basic_string<boost::uint32_t> expr_str = "Активы";
  u32regex r = make_u32regex("активы", regex::perl | regex::icase);
  return u32regex_search(text, r) ? 0 : 1;
}
```

```
% nix develop
% clang++ boost_regex_unicode_runs.cpp
In file included from boost_regex_unicode_runs.cpp:1:
In file included from /nix/store/a7s0kbig4a5s6237h2jjgwrr1zis0006-boost-1.89.0-dev/include/boost/regex/icu.hpp:23:
In file included from /nix/store/a7s0kbig4a5s6237h2jjgwrr1zis0006-boost-1.89.0-dev/include/boost/regex/v5/icu.hpp:25:
In file included from /nix/store/cdxfm0j5g928489bz7kswgbwybnmk32k-icu4c-76.1-dev/include/unicode/coll.h:61:
In file included from /nix/store/3dwjv57wm7w0dl8fqz4x9dczzflwsny4-libcxx-19.1.7-dev/include/c++/v1/functional:552:
In file included from /nix/store/3dwjv57wm7w0dl8fqz4x9dczzflwsny4-libcxx-19.1.7-dev/include/c++/v1/__functional/boyer_moore_searcher.h:27:
In file included from /nix/store/3dwjv57wm7w0dl8fqz4x9dczzflwsny4-libcxx-19.1.7-dev/include/c++/v1/vector:325:
In file included from /nix/store/3dwjv57wm7w0dl8fqz4x9dczzflwsny4-libcxx-19.1.7-dev/include/c++/v1/__format/formatter_bool.h:19:
In file included from /nix/store/3dwjv57wm7w0dl8fqz4x9dczzflwsny4-libcxx-19.1.7-dev/include/c++/v1/__format/formatter_integral.h:21:
In file included from /nix/store/3dwjv57wm7w0dl8fqz4x9dczzflwsny4-libcxx-19.1.7-dev/include/c++/v1/__format/formatter_output.h:22:
In file included from /nix/store/3dwjv57wm7w0dl8fqz4x9dczzflwsny4-libcxx-19.1.7-dev/include/c++/v1/__format/parser_std_format_spec.h:39:
/nix/store/3dwjv57wm7w0dl8fqz4x9dczzflwsny4-libcxx-19.1.7-dev/include/c++/v1/string:820:42: error: implicit instantiation of undefined template 'std::char_traits<unsigned int>'
  820 |   static_assert(is_same<_CharT, typename traits_type::char_type>::value,
      |                                          ^
boost_regex_unicode_runs.cpp:7:38: note: in instantiation of template class 'std::basic_string<unsigned int>' requested here
    7 |   std::basic_string<boost::uint32_t> expr_str = "Активы";
      |                                      ^
/nix/store/3dwjv57wm7w0dl8fqz4x9dczzflwsny4-libcxx-19.1.7-dev/include/c++/v1/__fwd/string.h:23:29: note: template is declared here
   23 | struct _LIBCPP_TEMPLATE_VIS char_traits;
      |                             ^
boost_regex_unicode_runs.cpp:7:38: error: no viable conversion from 'const char[13]' to 'std::basic_string<boost::uint32_t>' (aka 'basic_string<unsigned int>')
    7 |   std::basic_string<boost::uint32_t> expr_str = "Активы";
      |                                      ^          ~~~~~~~~
/nix/store/3dwjv57wm7w0dl8fqz4x9dczzflwsny4-libcxx-19.1.7-dev/include/c++/v1/string:999:71: note: candidate constructor not viable: no known conversion from 'const char[13]' to 'const basic_string<unsigned int> &' for 1st argument
  999 |   _LIBCPP_CONSTEXPR_SINCE_CXX20 _LIBCPP_STRING_INTERNAL_MEMORY_ACCESS basic_string(const basic_string& __str)
      |                                                                       ^            ~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/3dwjv57wm7w0dl8fqz4x9dczzflwsny4-libcxx-19.1.7-dev/include/c++/v1/string:1019:55: note: candidate constructor not viable: no known conversion from 'const char[13]' to 'basic_string<unsigned int> &&' for 1st argument
 1019 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX20 basic_string(basic_string&& __str)
      |                                                       ^            ~~~~~~~~~~~~~~~~~~~~
/nix/store/3dwjv57wm7w0dl8fqz4x9dczzflwsny4-libcxx-19.1.7-dev/include/c++/v1/string:1058:55: note: candidate constructor template not viable: no known conversion from 'const char[13]' to 'const unsigned int *' for 1st argument
 1058 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX20 basic_string(const _CharT* __s)
      |                                                       ^            ~~~~~~~~~~~~~~~~~
/nix/store/3dwjv57wm7w0dl8fqz4x9dczzflwsny4-libcxx-19.1.7-dev/include/c++/v1/string:1198:55: note: candidate constructor not viable: no known conversion from 'const char[13]' to 'initializer_list<unsigned int>' for 1st argument
 1198 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX20 basic_string(initializer_list<_CharT> __il)
      |                                                       ^            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/3dwjv57wm7w0dl8fqz4x9dczzflwsny4-libcxx-19.1.7-dev/include/c++/v1/string:989:64: note: explicit constructor is not a candidate
  989 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX20 explicit basic_string(const allocator_type& __a)
      |                                                                ^
/nix/store/3dwjv57wm7w0dl8fqz4x9dczzflwsny4-libcxx-19.1.7-dev/include/c++/v1/string:1154:93: note: explicit constructor is not a candidate
 1154 |   _LIBCPP_METHOD_TEMPLATE_IMPLICIT_INSTANTIATION_VIS _LIBCPP_CONSTEXPR_SINCE_CXX20 explicit basic_string(const _Tp& __t)
      |                                                                                             ^
2 errors generated.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -247,7 +247,6 @@ using namespace boost;
 
 int main() {
   std::string text = \"Активы\";
-  std::basic_string<boost::uint32_t> expr_str = \"Активы\";
   u32regex r = make_u32regex(\"активы\", regex::perl | regex::icase);
   return u32regex_search(text, r) ? 0 : 1;
 }" BOOST_REGEX_UNICODE_RUNS)

